src/game/component/PhysicsComponent.h

```diff
@@ -24,6 +24,7 @@ class PhysicsComponent : public IComponent
 	void IncrementVelocity(glm::vec3 velocity){ m_Velocity += velocity; }
 
 	void SetGravity(bool on){ m_IsGravityOn = on; }
+	const glm::vec3 GetVelocity() { return m_Velocity; }
```

Comment:
Minor nitpick: can you make this a const function? i.e. GetVelocity() const
It'll make it easier if the underlying EntitySystem ever needs to distinguish between const and non-const components later on.